lib/_emerge/depgraph.py

```diff
@@ -9339,6 +9341,15 @@ def find_smallest_cycle(mergeable_nodes, local_priority_range):
                             if gather_deps(
                                 priority, mergeable_nodes, selected_nodes, node
                             ):
+                                if len(selected_nodes) == 1 and asap_mode:
```

Comment:
Thanks. I should've really thought of this before when we spoke about it. We should still obviously think about proper priorities but this is good enough.